crypto/crypto.go

```diff
@@ -166,17 +165,28 @@ func HexToECDSA(hexkey string) (*ecdsa.PrivateKey, error) {
 
 // LoadECDSA loads a secp256k1 private key from the given file.
 func LoadECDSA(file string) (*ecdsa.PrivateKey, error) {
-	buf := make([]byte, 64)
-	fd, err := os.Open(file)
+	stat, err := os.Stat(file)
 	if err != nil {
 		return nil, err
 	}
-	defer fd.Close()
-	if _, err := io.ReadFull(fd, buf); err != nil {
+	size := stat.Size()
+	// Allow two extra chars for possible line ending to be checked later
+	if size < 64 || size > 66 {
+		return nil, fmt.Errorf("expected 64 bytes, got %v", size)
+	}
+	buf, err := ioutil.ReadFile(file)
+	if err != nil {
 		return nil, err
 	}
+	// Check line ending
+	maybeLineEnding := buf[64:]
+	for _, ch := range maybeLineEnding {
+		if ch != '\n' && ch != '\r' {
+			return nil, fmt.Errorf("expected 64 bytes, got %v", size)
+		}
+	}
```

Comment:
I think you can remove that entire thing, and just do 
```
buf := bytes.TrimSpace(buf)
if len(buf) != 64 {
  return nil, fmt.Errorf("expected 64 bytes, got %v", len(buf))
}
key, err := hex.DecodeString(string(buf))
```

Reply:
Thanks. Done.